File: DSA-Python/5-array/1-largest-element.py


# sort - O(n. log n), O(n)
def largest_element_1(nums:list) ->int:
    nums.sort()
    
    return nums[-1]
    


# optimal - O(n), O(1)
def largest_element_2(nums: list) -> int:
    return max(nums)


if __name__ == "__main__": 
    loops = int(input())

    for _ in range(loops):
        nums = list(map(int, input().split()))
        
        res = largest_element_1(nums)
        res = largest_element_2(nums)
        
        print(res)
        
        
'''
5
3 12 2 6 24 43
-2 45 54 3 12 23
-4 -1 -3 -5 -14
0 -1 -4 -4 -4 -4 0
1 2 3 4 5
'''

File: DSA-C++/6-array/1-largest-element.cpp
#include <bits/stdc++.h>

using namespace std;
// Given an array, return the largest element in the array.



// sort - O(n. logn), O(1)
int largest_element_1(vector <int> &nums) {
    sort(nums.begin(), nums.end());

    return nums[nums.size()-1];
}

// optimal - O(n), O(1)
int largest_element_2(vector <int> &nums) {
    int ans = nums[0];

    for (int &x:nums) {
        if (x > ans) {
            ans = x;
        }
    }

    return ans;
}

int main() {
    vector <int> nums = {8, 10, 5, 7, 9};
    nums = {2, 5, 1, 3, 0};

    cout << largest_element_1(nums) << endl;

    cout << largest_element_2(nums) << endl;

    return 0;
}